draft-ietf-ppm-dap.md

```diff
@@ -1186,27 +1271,157 @@ the last round is reached and each Aggregator computes an output share. All
 told there there are `ceil((ROUNDS+1)/2)` HTTP requests sent, where `ROUNDS` is
 the number of rounds specified by the VDAF.
 
-[OPEN ISSUE: For the moment we don't have an example of a VDAF that requires
-more than two rounds. Eventually we may want to specialize the spec to
-accommodate 1-2 rounds (or even just 1).]
+The state machine for VDAF preparation is shown in {{vdaf-prep-state-machine}}.
+
+~~~ state
+ START --> CONTINUED(round, host_prep_state)
+  |                |                    |
+  |                |                    v
+  +--> REJECTED <--+     FINISHED(out_share)
+~~~
+{: #vdaf-prep-state-machine title="State machine for VDAF preparation."}
+
+State transitions are made when the state is acted upon by the host's local
+inputs and/or a prep step sent by the peer. The initial state is `START`.
+
+~~~ transition
+State START
+
+Input
+
+  vdaf_verify_key: VDAF verification key associated with the task
+  is_leader: boolean indicating whether the host is the Leader
+  agg_param: aggregation parameter for the aggregation job
+  report_id: report ID
+  public_share: report public share
+  host_input_share: host's input share for the report
+
+def VdafPrepInit(vdaf_verify_key,
+                 is_leader,
+                 agg_param,
+                 report_id,
+                 public_share,
+                 host_input_share):
+  try:
+    host_prep_state = Vdaf.prep_init(
+      vdaf_verify_key,
+      0 if is_leader else 1,
+      agg_param,
+      report_id,
+      public_share,
+      host_input_share,
+    )
+    state CONTINUED(0, host_prep_state)
+  except:
+    state REJECTED
+~~~
+
+If the state is `REJECTED`, then processing halts. Otherwise, if the state is
+`CONTINUE`, then processing continues using the following transition rule until
+a terminal state is reached.
+
+~~~ transition
+State CONTINUED(round, host_prep_state)
+
+Input
+
+  is_leader: boolean indicating whether the host is the Leader
+  agg_param: aggregation parameter for the aggregation job
+  inbound: PrepareStep (or None)
+
+Output
+
+  outbound: PrepareStep (or None)
+
+def VdafPrepNext(is_leader, agg_param, inbound):
+  if type(inbound) == PrepareStep.initialize:
+    prep_msg, peer_prep_share = None, inbound.prep_share
+  elif type(inbound) = PrepareStep.continue:
+    prep_msg, peer_prep_share = inbound.prep_msg, inbound.prep_share
+  elif type(inbound) = PrepareStep.finish:
+    prep_msg, peer_prep_share = inbound.prep_msg, None
+  else:
+    prep_msg, peer_prep_share = None, None
```

Comment:
Should this be an error? I'm not sure we want to pass a `finished` or `reject` PrepareStep to this method.

Reply:
The first time the Leader calls `VdafPrepNext()` it is immediately after calling `VdafPrepInit()`. It has not received a prep_msg or prep share from its peer, so there's nothing to do. The outbound message on this call is the leader's first prep share.